### DISCUSSION TITLE
关于插件hmr的一些问题请教

### DISCUSSION THREAD

**Discussion by apades** [BewlyBewly/BewlyBewly#1161] [General]
我最近在探索怎么最好的开发插件方案，我[这里](https://github.com/apades/dmMiniPlayer/blob/main/assets/lib/entry-all-frames.js)勉强用的import打包后的js实现了刷新页面能更新ui，但体验还是不太好。我看到[项目里](https://github.com/BewlyBewly/BewlyBewly/blob/main/vite-mv3-hmr.ts)有用上vite的hmr模块，但实际更改了ui还是得去chrome://extensions/里面刷新插件才能更新，那这个hmr模块具体是干嘛来着

**Comment by hakadao** [comment]
之前是按照 https://github.com/antfu-collective/vitesse-webext 裏面的 mv3 hmr 分支改造的，那時主分支還沒有用 mv3 hmr 就用了分支上的，但是現在看好像分支已經刪了

然後後面我也測試一下之後的 vitesse-webext 的 main 分支，content script 修改也是不能及時替換，只有 options 頁面和 popup 可以

最後我對這部分底層的東西真的不懂，如果有更好的做法歡迎提出

**Comment by apades** [comment]
草，我刚才[随便试了下](https://github.com/apades/ext-vite-hmr-test)还真搞出了vite的hmr自动更新cs🤣不过现在修改都会触发vite强制刷新页面的，还没有不刷新就热重载的情况

**Comment by apades** [comment]
现在加了个@vitejs/plugin-react，带上它注入到html的script，还真实现了不刷新页面的hmr了😆

**Reply by hakadao** [reply]
之後試試你的做法，大佬厲害🫡
